lib/vfs/src/union_fs.rs

```diff
@@ -0,0 +1,1078 @@
+//! Another implementation of the union that uses paths,
+//! its not as simple as TmpFs. not currently used but was used by
+//! the previoulsy implementation of Deploy - now using TmpFs
```

Comment:
Do we need to keep this one if it's not used?

Reply:
I think itmight be useful to have specially for running on the frontend

Reply:
Let's keep it for now.

Reply:
Let's keep it for now.

Reply:
I'm cheekily going to resolve, reopen if you disagree.